src/scripts/main.js

```diff
@@ -1,7 +1,224 @@
+    row.forEach((value, colInd) => {
+      const cell = curCells[rowInd * boardArray.length + colInd];
+
+      if (value > 0) {
+        cell.textContent = value;
+        cell.classList = 'field-cell' + ` field-cell--${value}`;
+      } else {
+        cell.textContent = null;
+        cell.classList = 'field-cell';
+      }
+    });
+  });
+};
+
+const changeButton = () => {
+  startButton.classList += ' restart';
+  startButton.innerText = 'Restart';
+  startButton.style = 'border: 2px solid red; color: #776e65; outline: none;';
+  message.Start.classList.add('hidden');
+};
+
+document.addEventListener('keyup', (e) => {
+  if (!allowedKeys.includes(e.key) || isWin) {
+    return;
+  }
+
+  const prevBoard = JSON.parse(JSON.stringify(board));
+
+  switch (e.key) {
+    case 'ArrowLeft':
+      slideLeft();
+      getWinMessage(board);
+      getLoseMessage(board);
+      break;
+
+    case 'ArrowRight':
+      slideRight();
+      getWinMessage(board);
+      getLoseMessage(board);
+      break;
+
+    case 'ArrowUp':
+      slideUp();
+      getWinMessage(board);
+      getLoseMessage(board);
+      break;
+
+    case 'ArrowDown':
+      slideDown();
+      getWinMessage(board);
+      getLoseMessage(board);
+      break;
+
+    default:
+      break;
+  }
+
+  document.querySelector('.game-score').innerText = score;
+
+  if (isMoved(prevBoard)) {
+    addRandomField(board);
+    drawBoard(board, cells);
+  }
+});
+
+const isMoved = (prevBoard) => {
+  for (let r = 0; r < SIZE_BOARD; r++) {
+    for (let c = 0; c < SIZE_BOARD; c++) {
+      if (prevBoard[r][c] !== board[r][c]) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+const slideLeft = () => {
+  board = board.map((row) => slide(row));
+};
+
+const slideRight = () => {
+  board = board.map((row) => slide(row.reverse()).reverse());
+};
+
+function slideUp() {
+  const columns = prepareColumns(board);
+
+  board = prepareColumns(columns.map((col) => slide(col)));
+}
+
+function slideDown() {
+  let columns = prepareColumns(board);
+
+  columns = columns.map((col) => slide(col.reverse()).reverse());
+
+  board = prepareColumns(columns);
+}
+
+const deleteZeros = (row) => row.filter(Boolean);
+
+const slide = (row) => {
+  const rowClear = deleteZeros(row);
+  let newRow = rowClear;
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+      score += newRow[i];
+    }
+  }
+  newRow = deleteZeros(newRow);
+
+  while (newRow.length < 4) {
+    newRow.push(0);
+  }
+
+  return newRow;
+};
+
+function prepareColumns(boardArr) {
+  return boardArr.map((row, rowInd, rowArr) => {
+    return row.map((_, colInd) => {
+      return rowArr[colInd][rowInd];
+    });
+  });
+}
+
+const getWinMessage = (boardArray) => {
+  boardArray.forEach((row) => {
+    row.forEach((cell) => {
+      if (cell >= 2048) {
+        isWin = true;
+        message.Start.classList.remove('hidden');
```

Comment:
The win message is being shown by removing the 'hidden' class from 'message.Start'. This might be confusing as 'message.Start' is intended for the start message. Consider using 'message.Win' to display the win message instead.